src/server/scripts/EasternKingdoms/Karazhan/boss_shade_of_aran.cpp

```diff
@@ -503,6 +529,31 @@ class boss_shade_of_aran : public CreatureScript
                     return;
                 }
         }
+
+        void MoveInLineOfSight(Unit* who) override
+        {
+            ScriptedAI::MoveInLineOfSight(who);
+
+            if (SeenAtiesh || me->IsInCombat() || me->GetDistance2d(who) > me->GetAttackDistance(who) + 10.0f)
+                return;
+
+            Player* player = who->ToPlayer();
+            if (!player)
+                return;
+
+            for (auto id : AtieshStaves)
+            {
+                if (!player->HasItemOrGemWithIdEquipped(id, 1))
```

Comment:
HasItemOrGemWithIdEquipped has to iterate all equipment slots and their gems, wouldnt it be better to just take the main hand weapon and check the entry?